src/coord.jl

```diff
@@ -1 +1,157 @@
+the input object.
+"""
+function normalize_axis_order!(pj::Ptr{PJ}; ctx = C_NULL)
+    pj_for_gis = proj_normalize_for_visualization(pj, ctx)
+    proj_destroy(pj)
+    return pj_for_gis
+end
+
+function Base.inv(
+    trans::Transformation;
+    area::Ptr{PJ_AREA} = C_NULL,
+    ctx::Ptr{PJ_CONTEXT} = C_NULL,
+    always_xy::Bool = false,
+)
+    target_crs = proj_get_source_crs(trans.pj)
+    source_crs = proj_get_target_crs(trans.pj)
+    return Transformation(source_crs, target_crs; area=area, ctx=ctx, always_xy=always_xy)
+end
+
+function (trans::Transformation)(coord::StaticVector{2,<:AbstractFloat})
+    T = similar_type(coord)
+    coord = SVector{4, Float64}(coord[1], coord[2], 0.0, Inf)
+    p = proj_trans(trans.pj, PJ_FWD, coord)
+    return T(p[1], p[2])
+end
+
+function (trans::Transformation)(coord::StaticVector{3,<:AbstractFloat})
+    T = similar_type(coord)
+    coord = SVector{4, Float64}(coord[1], coord[2], coord[3], Inf)
+    p = proj_trans(trans.pj, PJ_FWD, coord)
+    return T(p[1], p[2], p[3])
+end
+
+function (trans::Transformation)(coord::StaticVector{4,<:AbstractFloat})
+    T = similar_type(coord)
+    coord = SVector{4, Float64}(coord[1], coord[2], coord[3], Inf)
+    p = proj_trans(trans.pj, PJ_FWD, coord)
+    return T(p)
+end
+
+function (trans::Transformation)(coord)::Coord234
+    # avoid splatting for performance
+    n = length(coord)
+    coord = if n == 2
+        proj_coord(coord[1], coord[2])
+    elseif n == 3
+        proj_coord(coord[1], coord[2], coord[3])
+    elseif n == 4
+        proj_coord(coord[1], coord[2], coord[3], coord[4])
+    else
+        throw(ArgumentError("input should be length 2, 3 or 4"))
+    end
+
+    p = proj_trans(trans.pj, PJ_FWD, coord)
+
+    if n == 2
+        return SVector{2, Float64}(p[1], p[2])
+    elseif n == 3
+        return SVector{3, Float64}(p[1], p[2], p[3])
+    else
+        return p
+    end
+end
+
+function CoordinateTransformations.compose(
+    trans1::Transformation,
+    trans2::Transformation;
+    area::Ptr{PJ_AREA} = C_NULL,
+    ctx::Ptr{PJ_CONTEXT} = C_NULL,
+    always_xy::Bool = false,
+)
+    # create a new Transformation from trans1 source to trans2 target
+    # can also be typed as trans1 ∘ trans2, typed with \circ
+    # a → b ∘ c → d doesn't make much sense if b != c, though we don't enforce it
+    source_crs = proj_get_source_crs(trans1.pj)
+    target_crs = proj_get_target_crs(trans2.pj)
+    return Transformation(source_crs, target_crs; area=area, ctx=ctx, always_xy=always_xy)
+end
+
+"""
+    enable_network(active::Bool = true, ctx::Ptr{PJ_CONTEXT} = C_NULL)::Bool
+
+Enable PROJ network access, if `active` is true, disable it if it is false. Optionally pass
+a context to set it for that context, instead of the global one.
+
+Returns true if network access is possible.
+"""
+function enable_network(active::Bool = true, ctx::Ptr{PJ_CONTEXT} = C_NULL)
```

Comment:
(optional) consider breaking this out into two functions to prevent any risk of confusion: `enable_network!(...)` and `disable_network!(...)`.

Reply:
I think I'd prefer keeping it together in a single function, `enable_network!(false)` meaning `disable_network!()` doesn't seem too strange to me.

But you're right that this needs a !: 625db851d3c20468f3dfa07c3790ad306df14df0